targets/targets.json

```diff
@@ -1439,7 +1439,7 @@
             "PSA"
         ],
         "is_disk_virtual": true,
-        "macros": ["CPU_MK64FN1M0VMD12", "FSL_RTOS_MBED"],
+        "macros": ["CPU_MK64FN1M0VMD12", "FSL_RTOS_MBED", "MBEDTLS_PSA_CRYPTO_C"],
```

Comment:
This would force all K64F users to enable PSA Crypto.

We don't want to turn on PSA for everybody's K64F. Memory usage and code size may be higher, for instance. If someone's application barely fit before, it won't fit if we force them to use PSA. Unfortunately, it looks like #8730 enables PSA by default on K64F. This is likely unintentional or a mistake.

PSA should be opt-in on K64F and come from the user's application (mbed_app.json), which means `MBEDTLS_PSA_CRYPTO_C` should not get turned on for a user unless they explicitly request it via their application's configuration.

Reply:
I've verified code size and static RAM usage both with and without PSA for blinky. LTO seems to be working, in that what the app doesn't use won't bloat the image.


With PSA and PSA Crypto

```
Total Static RAM memory (data + bss): 12672(+12672) bytes
Total Flash memory (text + data): 61565(+61565) bytes

Image: ./BUILD/K64F/GCC_ARM/mbed-os-example-blinky.bin
```

Without PSA and without PSA Crypto
```
Total Static RAM memory (data + bss): 12672(+12672) bytes
Total Flash memory (text + data): 61565(+61565) bytes

Image: ./BUILD/K64F/GCC_ARM/mbed-os-example-blinky.bin
```

So, I'm OK with this being available by default from a technical perspective.